pkg/credentialprovider/provider.go

```diff
@@ -121,3 +121,24 @@ func (d *CachingDockerConfigProvider) Provide() DockerConfig {
 	d.expiration = time.Now().Add(d.Lifetime)
 	return d.cacheDockerConfig
 }
+
+// Reset expiration so that the provider will get configure from real provider instead of cache.
+// This is useful in node e2e testings.
+func (d *CachingDockerConfigProvider) resetExpiration() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	d.expiration = time.Time{}
+}
+
+// ResetDefaultDockerProviderExpiration resets .dockercfg provider's expiration.
+func ResetDefaultDockerProviderExpiration() {
```

Comment:
Who is using this?

Reply:
Sorry, my mistake, forgot to remove this. It meant to reset DefaultDockerProviderExpiration, but then I found that kubelet and e2e are not in same process, so it won't take effect any all. So the config.json is created under `KubeletRootDirectory` finally.